### PR TITLE
Improve Windows building

### PR DESCRIPTION
**Pull request by patricksnape**
Add patch for VS2010 to find the correct path to VC on appveyor. Also,
improve the `bld.bat` for the new conda-build logic.

### CODE REVIEW

**Comment by patricksnape** [comment]
@jakirkham How do I kick off appveyor?

**Comment by jakirkham** [comment]
Yeah, you need to re-render. Here are the steps (assuming you are in top level directory of this repo).

```
conda install -y -n root -c conda-forge conda-smithy
conda smithy rerender
git add .
git commit -m "MNT: Re-rendered with conda-smithy <version>."
```

**Comment by jakirkham** [comment]
Sorry, `<version>` was meant to be a fill-in-the-blank. My bad for not specifying that. 😆

**Comment by jakirkham** [comment]
Probably is `0.10.3` if you are on latest.

**Comment by jakirkham** [comment]
Only one of the Windows builds errored. If you guessed Python 3.4 64-bit, you guessed correctly. 🎰 💰

Looks like it is having trouble finding the compiler.

**Comment by patricksnape** [comment]
Visual Studio 2010 you are the bane of my life.

**Comment by patricksnape** [comment]
@jakirkham :open_mouth:

**Comment by jakirkham** [comment]
Nice!

**Comment by patricksnape** [comment]
Ok well this resembles what I have on menpo now and seems to work. Everyone happy to merge this?

**Comment by 183amir** [comment]
I don't think you need to change the build number if you are adding Windows builds.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,22 +13,22 @@ source:
...
-  skip: True  # [win]
 
 requirements:
   build:
     - python  # [win]
 
 test:
   requires:
-    - python {{ environ['PY_VER'] + '*' }}                # [win]
+    - python                                              # [win]
```
Comment (jakirkham) [inline recipe/meta.yaml:31]: It works without the version specified? I'm a bit worried about this. Couple options here.

Could try using this

```
test:
    requires:
        - python 2.7.*  # [win and py27]
        - python 3.4.*  # [win and py34]
        - python 3.5.*  # [win and py35]
```

Alternatively, could try unpinning `conda-build`. I believe all the current bugs are fixed. Though don't know if there are new ones or not yet.

Comment (patricksnape) [inline recipe/meta.yaml:31]: I did remove the conda-build pinning to get it to build in the first place. I though the same at first but it appears to work now... (conda-build == 1.20.3)

**Comment by patricksnape** [comment]
@jakirkham With regards to build number - will a new Linux version get uploaded if the version number is the same? Or does conda-smithy recognise the build is identical and just skip uploading?

File: recipe/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipe/meta.yaml:31]: > I did remove the conda-build pinning to get it to build in the first place.

Ah, I see. I guess it is hard to tell as `appveyor.yml` was added in this PR too.

> I though the same at first but it appears to work now...

Still makes me a little nervous, but it is not a blocker from me. This only affects tests not the package itself.

**Comment by jakirkham** [comment]
LGTM

**Comment by jakirkham** [comment]
> With regards to build number - will a new Linux version get uploaded if the version number is the same? Or does conda-smithy recognise the build is identical and just skip uploading?

Sorry, that didn't show up when I commented. Yes, it will upload a new package for *NIXes. It probably isn't needed as *NIX builds are unchanged here. Though those CI builds will happen anyways. Honestly, it isn't a problem for me either way.

**Pull request merged**